### ISSUE TITLE
Custom username & password requirement

### ISSUE THREAD

**Issue by zaniar** [apioo/fusio#68]
We should allowed to configure the requirement of username & password like character length & character allowed.

Two items on Setting where we can enter regex would be good.

**Comment by k42b3** [comment]
Currently those requirements are [hardcoded](https://github.com/apioo/fusio-impl/blob/master/src/Service/User/ValidatorTrait.php#L76) in the user service. I think it is a good idea to make the password requirements configureable. But I would rather not use a regexp for this instead provide concrete settings to configure the pw validator i.e.:

```
user_pw_length  = 8
user_pw_alpha   = 1
user_pw_numeric = 1
user_pw_special = 1
```

Then the user could configure the min count of each char type. This is easier and protects the user from writing a insecure regexp.

Regarding the username I would like to keep this validation to the system. Which special chars do you need in the username which are currently not allowed?

**Comment by zaniar** [comment]
Currently, I do not have problem with username, since I do not need it for now. I give it the same value as email.

My current issue is with password. I want to allow alpha character only password.

My current hack is appending numeric & special character to password inputted by user.

Concrete settings instead of regex sounds good for me.
I do not think we need special pattern for password, like password must contains a certain word.

**Comment by k42b3** [comment]
Ok, I think the best solution is to simply allow all characters in the password and not to force a special or numeric character. Its only important that the password has at least 8 signs. This would solve your problem and the user can still choose a save password. The next release will also include a simple config `user_pw_length` where it is possible to set the min required length for the password which is default 8 signs.

**Comment by k42b3** [comment]
This is implemented in the 0.9.9 release so you are not longer required to provide a special char and you can configure the min pw length default is 8 characters.
